includes/content-distribution/class-cap-authors.php

```diff
@@ -0,0 +1,196 @@
+<?php
+/**
+ * Newspack Network author ingestion for content distribution.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack_Network\Content_Distribution;
+
+use Newspack_Network\Debugger;
+use Newspack_Network\User_Update_Watcher;
+use WP_Error;
+use WP_Post;
+
+/**
+ * Class to handle author ingestion for content distribution.
+ */
+class Cap_Authors {
+
+	/**
+	 * TODO
+	 */
+	const AUTHOR_LIST_META_KEY = 'newspack_network_author_list';
+
+	/**
+	 * Get things going.
+	 *
+	 * @return void
+	 */
+	public static function init(): void {
+		if ( ! self::is_co_authors_plus_active() ) {
+			return;
+		}
+		Cap_Authors_Filters::init();
+
+		add_action( 'set_object_terms', [ __CLASS__, 'handle_cap_author_change' ], 10, 6 );
+	}
+
+	/**
+	 * Helper to check if Co-Authors Plus is active.
+	 *
+	 * @return bool Whether Co-Authors Plus is active.
+	 */
+	public static function is_co_authors_plus_active(): bool {
+		global $coauthors_plus;
+
+		return $coauthors_plus instanceof \CoAuthors_Plus && function_exists( 'get_coauthors' );
+	}
+
+	/**
+	 * Action callback.
+	 *
+	 * Add a postmeta entry with the Co-Authors Plus authors for outgoing posts.
+	 *
+	 * @param int    $object_id The object ID.
+	 * @param array  $terms The terms.
+	 * @param array  $tt_ids The term taxonomy IDs.
+	 * @param string $taxonomy The taxonomy.
+	 * @param bool   $append Whether to append.
+	 * @param array  $old_tt_ids The old term taxonomy IDs.
+	 *
+	 * @return void
+	 */
+	public static function handle_cap_author_change( $object_id, $terms, $tt_ids, $taxonomy, $append, $old_tt_ids ): void {
+		if ( 'author' !== $taxonomy ) { // Co-Authors Plus author taxonomy.
+			return;
+		}
+		// If the terms are the same, we don't need to do anything. Note that one array has string values and one has
+		// int values, so we use array_map with the intval for the comparison.
+		if ( array_map( 'intval', $old_tt_ids ) === array_map( 'intval', $tt_ids ) ) {
+			return;
+		}
+
+		try {
+			$outgoing_post = new Outgoing_Post( $object_id );
+			if ( ! $outgoing_post->is_distributed() ) {
```

Comment:
I don't really want to do this on all posts, but if I remove these lines the author change works a bit more consistently.

Reply:
That makes sense. I changed the code now to call distribute_partial so I need the outgoing post now.